src/game.cpp

```diff
@@ -11602,14 +11602,46 @@ void game::read()
     auto filter = []( const item &it ) {
         return it.is_book();
     };
-    int pos = inv_for_filter( _("Read:"), filter );
 
-    if (pos == INT_MIN) {
-        add_msg(_("Never mind."));
+    auto item_loc = inv_map_splice( filter, _("Read:") );
+
+    const int inv_pos = item_loc.get_inventory_position();
+
+    if ( inv_pos != INT_MIN ) {
+        draw();
+        u.read( inv_pos, FROM_INVENTORY );
+        return;
+    }
+
+    item *it = item_loc.get_item();
+    if ( it == nullptr ) {
+        add_msg( _("Never mind.") );
+        return;
+    }
+
+    if ( !u.can_pickup(true) ) {
+        return;
+    }
+
+    if ( !u.can_pickVolume( it->volume() ) ) {
+        add_msg( m_info, _("Can't pickup %s as exceeded allowed volume. Reading aborted."), it->display_name().c_str() );
```

Comment:
>Also, double space after end marks please.
Will try to but might miss out some.
However, find that too much space also make it unclear. Thus be changing to only first () have space.  eg   f1( p1, f2(ns1,ns2) )  {         <-- double space between )  {